tests/ZfcRbacTest/Factory/ControllerGuardFactoryTest.php

```diff
@@ -50,13 +50,17 @@ public function testFactory()
             $this->getMock('ZfcRbac\Service\RoleService', [], [], '', false)
         );
 
-        $pluginManager = new GuardPluginManager();
-        $pluginManager->setServiceLocator($serviceManager);
+        $pluginManager = new GuardPluginManager($serviceManager);
 
         $factory         = new ControllerGuardFactory();
-        $controllerGuard = $factory->createService($pluginManager);
 
-        $this->assertInstanceOf('ZfcRbac\Guard\ControllerGuard', $controllerGuard);
-        $this->assertEquals(GuardInterface::POLICY_ALLOW, $controllerGuard->getProtectionPolicy());
+        if (method_exists($serviceManager, 'build')) { // servicemanager v3
```

Comment:
why do you check this? 
```php
$guard = $factory->createService($pluginManager);
```
should be for both v2 and v3 versions the same or if you really want to test v3 should be:
```php
$guard = $factory($serviceManager, \ZfcRbac\Guard\ControllerGuard::class);
```
should be fixed in all tests